src/components/TodoItem.tsx

```diff
@@ -0,0 +1,216 @@
+    dispatch({ type: 'startUpdate' });
+    dispatch({ type: 'selectTodo', payload: todo.id });
+    if (updatedTitle.trim() === todo.title) {
+      setIsEditing(false);
+    }
+
+    if (updatedTitle.trim().length === 0) {
+      deleteTodo(todo.id)
+        .then(() => {
+          setIsEditing(false);
+          dispatch({ type: 'deleteTodo', payload: todo.id });
+        })
+        .catch(() =>
+          dispatch({ type: 'showError', payload: 'Unable to update a todo' }),
+        )
+        .finally(() => dispatch({ type: 'stopUpdate' }));
+    }
+
+    updateTodo(todo.id, { ...todo, title: updatedTitle.trim() })
+      .then(newTodo => {
+        dispatch({
+          type: 'updateTodo',
+          payload: { ...newTodo },
+        });
+
+        setIsEditing(false);
+      })
+      .catch(() => {
+        dispatch({ type: 'showError', payload: 'Unable to delete a todo' });
+        setIsEditing(true);
+      })
+      .finally(() => {
+        dispatch({ type: 'stopUpdate' });
+      });
+  };
+
+  const handleOnTextChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUpdatedTitle(e.target.value);
+  };
+
+  const handleOnInputBlur = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUpdatedTitle(e.target.value);
+    dispatch({ type: 'startUpdate' });
+    dispatch({ type: 'selectTodo', payload: todo.id });
+
+    if (updatedTitle.trim() === todo.title) {
+      setIsEditing(false);
+      dispatch({ type: 'stopUpdate' });
+
+      return;
+    }
+
+    if (updatedTitle.trim().length === 0) {
+      deleteTodo(todo.id)
+        .then(() => {
+          setIsEditing(false);
+          dispatch({ type: 'deleteTodo', payload: todo.id });
+        })
+        .catch(() =>
+          dispatch({ type: 'showError', payload: 'Unable to delete a todo' }),
+        )
+        .finally(() => dispatch({ type: 'stopUpdate' }));
+
+      return;
+    }
+
+    updateTodo(todo.id, { ...todo, title: updatedTitle.trim() })
+      .then(newTodo => {
+        dispatch({
+          type: 'updateTodo',
+          payload: { ...newTodo },
+        });
+
+        setIsEditing(false);
+      })
+      .catch(() => {
+        dispatch({ type: 'showError', payload: 'Unable to update a todo' });
+        setIsEditing(true);
+      })
+      .finally(() => {
+        dispatch({ type: 'stopUpdate' });
+      });
+  };
+
+  const handleEscKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Escape') {
+      setUpdatedTitle(todo.title);
+      setIsEditing(false);
+    }
+  };
+
+  //#endregion
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames('todo', { ['completed']: todo.completed })}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={handleOnCheckChange}
+        />
+      </label>
+
+      {isEditing && selectedTodo === todo.id ? (
```

Comment:
try to remove the logic `selectedTodo === todo.id` and deploy again the PR. Let's see if doing that the test pass